### PR TITLE
(PCP-672) Document broker resource usage

### PR DESCRIPTION
**Pull request by MikaelSmith**
See the ticket for detailed measurements that led to these
recommendations.

### CODE REVIEW

**Comment by nicklewis** [comment]
:+1:

**Comment by MikaelSmith** [comment]
ping @puppetlabs/pcp-broker-maintainers

**Comment by wkalt** [comment]
seems worth mentioning message overhead/lack of significance now that we have data.

**Comment by MikaelSmith** [comment]
Sure, I'll add that.

File: doc/resources.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+Resource usage of pcp-broker primarily means memory required by the JVM and
+file descriptors used. Usage scales linearly with number of active connections.
+
+Specifically, the pcp-broker requires ~40KB of memory (when restricted with the
+JVM option `-Xmx`) and 1 file descriptor per connection, with a baseline of
+~60MB and 200 file descriptors. For a deployment of 100 agents, expect to
+configure the JVM with at least `-Xmx64m` and 300 file descriptors.
+
+Message handling requires minimal additional memory.
```
Comment (MikaelSmith) [inline doc/resources.md]: I tried 40k messages and still didn't hit any issues, so I think we can pretty much ignore it for our expected message size.

**Pull request merged**